### PR TITLE
Versioning and Dependabot

### PR DESCRIPTION
**Pull request by TonyXiang8787**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -75,7 +76,7 @@ def generate_build_ext(pkg_dir: str, pkg_name: str):
     # include-folders
     include_dirs = [
```
Comment (bramstoeller) [inline setup.py:77]: Must these paths be strings, or can we use Path objects?

File: setup.py [hunk 2]
```diff
@@ -89,29 +90,29 @@ def generate_build_ext(pkg_dir: str, pkg_name: str):
         ("EIGEN_MPL2_ONLY", "1"),  # only MPL-2 part of eigen3
         ("POWER_GRID_MODEL_USE_MKL_AT_RUNTIME", 1),  # use mkl runtime loading
     ]
+    pkg_bin_dir = pkg_dir / "src" / pkg_name
 
     # remove old extension build
-    shutil.rmtree(os.path.join(pkg_dir, "build"), ignore_errors=True)
+    shutil.rmtree(pkg_dir / "build", ignore_errors=True)
```
Comment (bramstoeller) [inline setup.py:96]: Do we want to ignore errors? If we expect errors in some cases, it might be good to write some comments about the expected errors.

Comment (bramstoeller) [reply setup.py:96]: Suggestions:
```python
build_dir = pkg_dir / "build"
if build_dir.exists():
    shutil.rmtree(build_dir)
```

File: setup.py [hunk 3]
```diff
@@ -187,9 +178,14 @@ def convert_long_description(raw_readme: str):
         return re.sub(r"(\[[^\(\)\[\]]+\]\()((?!http)[^\(\)\[\]]+\))", f"\\1{url}\\2", raw_readme)
 
 
-def get_version(pkg_dir):
-    with open(os.path.join(pkg_dir, "VERSION")) as f:
+def get_version(pkg_dir: Path):
```
Comment (bramstoeller) [inline setup.py:181]: According to mypy it is recommended to add the return type as well when you use typehinting (even if the return type is None). So either use no type hinting, or full type hinting.

File: setup.py [hunk 4]
```diff
@@ -226,14 +246,15 @@ def build_pkg(setup_file, author, author_email, description, url):
...
     pkg_pip_name = "power-grid-model"
     pkg_name = pkg_pip_name.replace("-", "_")
-    with open(os.path.join(pkg_dir, "README.md")) as f:
+    with open(pkg_dir / "README.md") as f:
         long_description = f.read()
         long_description = convert_long_description(long_description)
-    with open(os.path.join(pkg_dir, "requirements.txt")) as f:
+    with open(pkg_dir / "requirements.txt") as f:
         required = f.read().splitlines()
         required = [x for x in required if "#" not in x]
```
Comment (bramstoeller) [inline setup.py:259]: It is possible to add comments to your pip package, e.g.
```pip
fancypackage==1.2.3  # Use version pin until bug #42 is solved
```

So I would recommend to use `if lstrip(x) and lstrip(x)[0] != "#"`

**Review by bramstoeller — APPROVED** [review]
Approved, but please review my comments before merging and decide if you want to alter some some lines or leave it as is.

File: setup.py (hunk 1 above)
Comment (TonyXiang8787) [reply setup.py:77]: It has to be string.

File: setup.py (hunk 4 above)
Comment (TonyXiang8787) [reply setup.py:262]: What should you import for `strip`?

Comment (bramstoeller) [reply setup.py]: Sorry, I meant `x.strip()`

**Review by bramstoeller — APPROVED** [review]

**Pull request merged**